src/lib/esnext.promise.d.ts

```diff
@@ -0,0 +1,16 @@
+interface PromiseConstructor {
+    /**
+     * Takes a callback of any kind (returns or throws, synchronously or asynchronously) and wraps its result
+     * in a Promise.
+     *
+     * @param callbackFn A function that is called synchronously. It can do anything: either return
+     * a value, throw an error, or return a promise.
+     * @param args Additional arguments, that will be passed to the callback.
+     *
+     * @returns A Promise that is:
+     * - Already fulfilled, if the callback synchronously returns a value.
+     * - Already rejected, if the callback synchronously throws an error.
+     * - Asynchronously fulfilled or rejected, if the callback returns a promise.
+     */
+    try<T, U extends unknown[]>(callbackFn: (...args: U) => T | PromiseLike<T>, ...args: U): Promise<T>;

```

Comment:
Looking at the type of `Promise.resolve` (which was improved in #33074), I think the return type should be `Promise<Awaited<T>>`, because of problems like #48195.

Reply:
Good suggestion, fixed!